### PR TITLE
chore: remove `TransactionLookup` stage

### PR DESCRIPTION
**Pull request by joshieDo**
With the removal of `TxHash` from the `Transactions` table, the `TransactionLookup` stage suffered a big performance hit, since once we read a transaction from the table, we have to calculate its transaction hash. 

As long as it doesn't really have any impact elsewhere (not that i can see), proposing that we just remove this stage, and insert the lookup, the moment we have access to the transaction and its hash during the bodies stage.

One side effect will be that the first `cursor.append` will result in the initial data not being as compacted as it was before if the block range for the bodies stage is smaller.

### CODE REVIEW

File: crates/stages/src/stages/bodies.rs [hunk 1]
```diff
@@ -102,6 +104,8 @@ impl<DB: Database, D: BodyDownloader> Stage<DB> for BodyStage<D> {
         trace!(target: "sync::stages::bodies", bodies_len = downloaded_bodies.len(), "Writing blocks");
 
         let mut highest_block = from_block;
+        let mut tx_hashes = Vec::with_capacity(downloaded_bodies.len() * 10);
```
Comment (joshieDo) [inline crates/stages/src/stages/bodies.rs:107]: open to suggestions

**Comment by joshieDo** [comment]
The difference between then and this current implementation, is that we collect all transaction hashes first (for this block range) and write them in bulk with a cursor instead of `.put()`.

An alternative could be to write them to a separate place (file or whatever), and write them to the db during the transaction lookup stage with `append`

**Comment by rkrasiuk** [comment]
syncing from this branch to see if there is any significant impact on the performance

**Comment by rkrasiuk** [comment]
Observed a significant impact on performance starting from block ~4.5mil, single iteration of the stage gradually increased up to 30s at block 5.5mil after which the process ran out of memory (64gb). Majority of the time is spent writing blocks to the database

**Comment by gakonst** [comment]
> An alternative could be to write them to a separate place (file or whatever), and write them to the db during the transaction lookup stage with append

This seems like it could be the best solution here, like a mini-ETL? Where else would we want to utilize this?

**Comment by joshieDo** [comment]
> like a mini-ETL? 
yes,

> Where else would we want to utilize this?
any table that has keys as hashes can probably benefit from this approach (HashedStorage, HashedAccount) since we can use `.append` over one big batch, resulting in smaller table size

**Comment by joshieDo** [comment]
> like a mini-ETL? 

yes,

> Where else would we want to utilize this?

any table that has keys as hashes can probably benefit from this approach (HashedStorage, HashedAccount) since we can use `.append` over one big batch, resulting in smaller table size.

If it's faster, we'll have to test it out though...

**Comment by joshieDo** [comment]
> like a mini-ETL? 

yes,


> Where else would we want to utilize this?

any table that has keys as hashes can probably benefit from this approach (HashedStorage, HashedAccount) since we can use `.append` over one big batch, resulting in smaller table size.

If it's faster, we'll have to test it out though... but i think so

**Pull request closed without merging**